plugin/storage/cassandra/spanstore/reader.go

```diff
@@ -35,26 +35,70 @@ import (
 )
 
 const (
+	bucketRange        = `(0,1,2,3,4,5,6,7,8,9)`
 	querySpanByTraceID = `
-		SELECT trace_id, span_id, parent_id, operation_name, flags, start_time, duration, tags, logs, refs, process 
-		FROM traces 
+		SELECT trace_id, span_id, parent_id, operation_name, flags, start_time, duration, tags, logs, refs, process
+		FROM traces
 		WHERE trace_id = ?`
+	queryByTag = `
+		SELECT trace_id
+		FROM tag_index
+		WHERE service_name = ? AND tag_key = ? AND tag_value = ? and start_time > ? and start_time < ?
+		LIMIT ?`
+	queryByServiceName = `
+		SELECT trace_id
+		FROM service_name_index
+		WHERE bucket IN ` + bucketRange + ` AND service_name = ? AND start_time > ? AND start_time < ?
+		LIMIT ?`
+	queryByServiceAndOperationName = `
+		SELECT trace_id
+		FROM service_operation_index
+		WHERE service_name = ? AND operation_name = ? AND start_time > ? AND start_time < ?
+		LIMIT ?`
```

Comment:
cqlsh:jaeger_v2_test> select * from service_operation_index  where service_name = 'someServiceName' and operation_name = 'someOperationName' and start_time > 0 and start_time < 1000 order by start_time;

 service_name | operation_name | start_time | trace_id
--------------+----------------+------------+----------

(0 rows)
cqlsh:jaeger_v2_test> 

you're right. Fixing.